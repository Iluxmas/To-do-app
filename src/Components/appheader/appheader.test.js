import React from "react";
import { unmountComponentAtNode, render } from "react-dom";
import { getByTestId, render as renderTL, cleanup } from "@testing-library/react";
import { act } from "react-dom/test-utils";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom/extend-expect";
import AppHeader from "./appheader";

afterEach(cleanup);

it("AppHeader renders without crash", () => {
  renderTL(<AppHeader toDo="1" done="1" />);
});

it("AppHeader renders props correctly", () => {
  const { getByTestId } = renderTL(<AppHeader toDo="3" done="5" />);
  expect(getByTestId("h2")).toHaveTextContent("3 more to do, 5 done");
});

it("AppHeader matches snapshot", () => {
  const tree = renderer.create(<AppHeader toDo="3" done="3" />).toJSON();
  expect(tree).toMatchSnapshot();
});
