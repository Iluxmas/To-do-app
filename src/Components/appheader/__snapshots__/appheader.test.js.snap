// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader matches snapshot 1`] = `
<div
  className="app__header d-flex"
>
  <h1>
    My Todo List
  </h1>
  <h2
    data-testid="h2"
  >
    3
     more to do, 
    3
     done
  </h2>
</div>
`;
